pkg/localstorev2/internal/chunkstore/transaction.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package chunkstore
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ethersphere/bee/pkg/sharky"
+	"github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// txSharky provides a simple txn functionality over the Sharky store.
+// It mainly exist to to support the chunk store Delete operation where
+// the Release calls are postponed until Commit or Rollback is called.
+type txSharky struct {
+	Sharky
+
+	opsMu         sync.Mutex
+	writtenLocs   []sharky.Location
+	toReleaseLocs map[[32]byte]sharky.Location
+	toReleaseSums map[sharky.Location][32]byte
+}
+
+func (t *txSharky) Write(ctx context.Context, buf []byte) (sharky.Location, error) {
+	var (
+		sum = sha256.Sum256(buf)
+		loc sharky.Location
+		err error
+	)
+
+	t.opsMu.Lock()
+	defer t.opsMu.Unlock()
+
+	loc, ok := t.toReleaseLocs[sum]
```

Comment:
So I cannot think of a better way at the moment. But lets add some comments here on why this is being done.

Reply:
Done.